test/e2e/resource_template_test.go

```diff
@@ -54,6 +54,45 @@ spec:
 		})
 }
 
+func (s *ResourceTemplateSuite) TestResourceTemplateWithPod() {
+	s.Given().
+		Workflow(`
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: k8s-resource-tmpl-with-pod-
+spec:
+  serviceAccount: argo
+  entrypoint: main
+  templates:
+    - name: main
+      serviceAccountName: argo
+      resource:
+        action: create
+        successCondition: status.phase == Succeeded
+        failureCondition: status.phase == Failed
+        manifest: |
+          apiVersion: v1
+          kind: Pod
+          metadata:
+            generateName: k8s-pod-resource-
+          spec:
+            serviceAccountName: argo
+            containers:
+            - name: pi
+              image: perl
```

Comment:
should not use a special image, makes builds slow, can you use a whitelisted image please, e.g `argoproj/argosay:v2`

Reply:
Sounds good. Done.

Reply:
Sounds good. Changed.